File: VB/dxSample/Custom Expression Editor Functions/CustomFormatFunction.vb
Imports DevExpress.Data.Filtering
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace dxSample
	Public Class CustomFormatFunction
		Implements ICustomFunctionOperatorBrowsable

		Public ReadOnly Property Category() As FunctionCategory
			Get
				Return FunctionCategory.Text
			End Get
		End Property
		Public ReadOnly Property Description() As String
			Get
				Return "CustomFormatFunction(string format, object arg0)" & vbCrLf & "Converts an arg0 value to a string based on a specified format"
			End Get
		End Property
		Public Function IsValidOperandCount(ByVal count As Integer) As Boolean
			Return count = 2
		End Function
		Public Function IsValidOperandType(ByVal operandIndex As Integer, ByVal operandCount As Integer, ByVal type As Type) As Boolean
			Return True
		End Function
		Public ReadOnly Property MaxOperandCount() As Integer
			Get
				Return 2
			End Get
		End Property
		Public ReadOnly Property MinOperandCount() As Integer
			Get
				Return 2
			End Get
		End Property
		Public Function Evaluate(ParamArray ByVal operands() As Object) As Object
			Dim res As String = String.Format(operands(0).ToString(), operands(1))
			Return res
		End Function
		Public ReadOnly Property Name() As String
			Get
				Return "CustomFormatFunction"
			End Get
		End Property
		Public Function ResultType(ParamArray ByVal operands() As Type) As Type
			Return GetType(String)
		End Function
	End Class
End Namespace

File: CS/dxSample/Custom Expression Editor Functions/CustomFormatFunction.cs
using DevExpress.Data.Filtering;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace dxSample {
    public class CustomFormatFunction : ICustomFunctionOperatorBrowsable {
        public FunctionCategory Category {
            get {
                return FunctionCategory.Text;
            }
        }
        public string Description {
            get {
                return "CustomFormatFunction(string format, object arg0)\r\nConverts an arg0 value to a string based on a specified format";
            }
        }
        public bool IsValidOperandCount(int count) {
            return count == 2;
        }
        public bool IsValidOperandType(int operandIndex, int operandCount, Type type) {
            return true;
        }
        public int MaxOperandCount {
            get {
                return 2;
            }
        }
        public int MinOperandCount {
            get {
                return 2;
            }
        }
        public object Evaluate(params object[] operands) {
            string res = String.Format(operands[0].ToString(), operands[1]);
            return res;
        }
        public string Name {
            get {
                return "CustomFormatFunction";
            }
        }
        public Type ResultType(params Type[] operands) {
            return typeof(string);
        }
    }
}
